src/Microsoft.EntityFrameworkCore.UnitOfWork/UnitOfWork.cs

```diff
@@ -103,7 +121,15 @@ public void ChangeDatabase(string database)
             var type = typeof(TEntity);
             if (!repositories.ContainsKey(type))
             {
-                repositories[type] = new Repository<TEntity>(_context);
+                if (_customRepositoryFactories != null && _customRepositoryFactories.ContainsKey(type))
```

Comment:
We cannot assume the custom repository only dependency DbContext. Maybe the better solution is:
1. First to ask ServiceProvider whether can fetch instance of IRepository or not;
2. If 1. false, we create one for it.

```csharp
public IRepository<TEntity> GetRepository<TEntity>() where TEntity : class
{
    if (repositories == null)
    {
        repositories = new Dictionary<Type, object>();
    }

    // what's the best way to support custom reposity?
    var customRepo = _context.GetService<IRepository<TEntity>>();
    if (customRepo != null)
    {
        return customRepo;
    }

    var type = typeof(TEntity);
    if (!repositories.ContainsKey(type))
    {
        repositories[type] = new Repository<TEntity>(_context);
    }

    return (IRepository<TEntity>)repositories[type];
}
```